File: src/test/java/application/InitTestServiceDecorator.java
package application;

import application.InitTestService;
import com.matag.game.status.GameStatus;

public class InitTestServiceDecorator extends InitTestService {
  private InitTestService initTestService;

  @Override
  public void initGameStatus(GameStatus gameStatus) {
    initTestService.initGameStatus(gameStatus);
  }

  public void setInitTestService(InitTestService initTestService) {
    initTestService.setCardInstanceFactory(this.getCardInstanceFactory());
    initTestService.setCards(this.getCards());
    this.initTestService = initTestService;
  }
}

File: src/test/kotlin/application/InitTestServiceDecorator.kt
package application

import com.matag.game.status.GameStatus

class InitTestServiceDecorator : InitTestService() {
    private var initTestService: InitTestService? = null

    override fun initGameStatus(gameStatus: GameStatus?) {
        initTestService!!.initGameStatus(gameStatus)
    }

    fun setInitTestService(initTestService: InitTestService) {
        initTestService.setCardInstanceFactory(this.getCardInstanceFactory())
        initTestService.setCards(this.cards)
        this.initTestService = initTestService
    }
}
